packages/discovery-provider/src/gated_content/content_access_checker.py

```diff
@@ -367,7 +367,20 @@ def _evaluate_conditions(
                 condition_options=condition_options,
             )
             if not has_access:
-                return False
+                # perhaps content was previously (not currently) purchase gated
+                # so we check if user had previously purchased content
+                if condition != "usdc_purchase":
```

Comment:
nit: should we have `usdc_purchase` as a constant/enum/type somewhere?

Reply:
we do. it's called `USDC_PURCHASE_KEY`. we should use it more!

Reply:
or rather, use it across